### PR TITLE
yesterday requirement

### PR DESCRIPTION
**Pull request by byau**

### CODE REVIEW

**Comment by byau** [comment]
up to 09 learnyounode

File: learnyounode/05-supplement.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    fs.readdir(directory, function (err, filelist) {
+        if (err) {
+            return callback(err);
+        }
+
+        var data = filelist.map(function (file) {
+            if ( path.extname(file) === '.' + ext ) {
+                return file;
+            }
+        });
```
Comment (GA-MEB) [inline learnyounode/05-supplement.js:15]: What is the purpose of using `.map` here? Why not just use `.filter`?

File: learnyounode/09-time-server.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+var net = require('net');
+
+var server = net.createServer(function(socket){
+  var date = new Date();
+  data = date.getFullYear()+'-'+('0'+(date.getMonth()+1)).slice(-2)+'-'+('0'+ date.getDate()).slice(-2)+' '+date.getHours()+':'+date.getMinutes() + '\n';
+  socket.end(data); // close the socket
+});
+server.listen(process.argv[2]);
```
Comment (GA-MEB) [inline learnyounode/09-time-server.js:8]: That approach to zero-fill is pretty unusual, but will work. However, you didn't apply it consistently - hours and minutes should really the same treatment.

**Comment by GA-MEB** [comment]
Nice work.

**Pull request closed without merging**